<a href="https://colab.research.google.com/github/KIISIISHERE/Implementation-of-Linear-Regression/blob/main/Implementation_of_Linear_Regression_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [50]:
from google.colab import files
files=files.upload()

In [21]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
#split the data into testing and training
from sklearn.model_selection import train_test_split
#regression Algorithm
from sklearn.linear_model import LinearRegression
#metrics
from sklearn.metrics import mean_squared_error,r2_score,classification_report

In [51]:
#load the dataset
data=pd.read_csv('/content/petrol_consumption.csv')

In [52]:
#check for null values
data.isnull().sum()

Petrol_tax                      0
Average_income                  0
Paved_Highways                  0
Population_Driver_licence(%)    0
Petrol_Consumption              0
dtype: int64

In [53]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48 entries, 0 to 47
Data columns (total 5 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Petrol_tax                    48 non-null     float64
 1   Average_income                48 non-null     int64  
 2   Paved_Highways                48 non-null     int64  
 3   Population_Driver_licence(%)  48 non-null     float64
 4   Petrol_Consumption            48 non-null     int64  
dtypes: float64(2), int64(3)
memory usage: 2.0 KB


In [54]:
#split the data into testing and training
X=data.drop('Petrol_Consumption',axis=1)
y=data['Petrol_Consumption']
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=42)

In [55]:
#Train the algorith
algorithm=LinearRegression()
algorithm.fit(X_train,y_train)

LinearRegression()

In [56]:
#use the algorithm to make prediction
petrol_predictions=algorithm.predict(X_test)
petrol_predictions

array([606.69266519, 673.77944169, 584.99149034, 563.53691024,
       519.05867235, 643.46100256, 572.89761422, 687.07703573,
       547.6093662 , 530.03762971])

In [57]:
#compare the predicted values vs actual values
df=pd.DataFrame({'Actual':y_test,'Predicted':petrol_predictions})
df


Actual   Predicted
27     631  606.692665
40     587  673.779442
26     577  584.991490
43     591  563.536910
24     460  519.058672
37     704  643.461003
12     525  572.897614
19     640  687.077036
4      410  547.609366
25     566  530.037630

In [58]:
#accuracy of the algorithm
print('Mean Squeared Error:',mean_squared_error(y_test,petrol_predictions))
print('R2 Score:',r2_score(y_test,petrol_predictions))

Mean Squeared Error: 4083.2558717453767
R2 Score: 0.3913664001428886


# ***Findings***
* ***MSE- How far off your predictions are from the actual values.***

* ***R2- the predicted values are 39% different from the actual values.***